src/main/java/com/artipie/auth/AuthFromYaml.java

```diff
@@ -65,22 +68,18 @@ public AuthFromYaml(final YamlMapping cred) {
     }
 
     @Override
-    public Optional<String> user(final String user, final String pass) {
+    public Optional<Authentication.User> user(final String user, final String pass) {
         final YamlMapping users = this.cred.yamlMapping("credentials");
-        Optional<String> res = Optional.empty();
+        Optional<Authentication.User> res = Optional.empty();
         //@checkstyle NestedIfDepthCheck (10 lines)
         if (users != null && users.yamlMapping(user) != null) {
             final String stored = users.yamlMapping(user).string("pass");
             if (stored != null) {
                 final String type = users.yamlMapping(user).string("type");
-                if (type != null) {
-                    if (check(stored, type, pass)) {
-                        res = Optional.of(user);
-                    }
-                } else {
-                    if (check(stored, pass)) {
-                        res = Optional.of(user);
-                    }
+                if (type != null && check(stored, type, pass)) {
+                    res = Optional.of(new User(user, AuthFromYaml.groups(users.yamlMapping(user))));
+                } else if (check(stored, pass)) {
+                    res = Optional.of(new User(user, AuthFromYaml.groups(users.yamlMapping(user))));
```

Comment:
@olenagerasimova merging of nested `if/else` statements you've mad are not identical. Is it intentional? I'd suggest to keep it as it is for now and address in separate issue/PR if it needs to be fixed